records/state/client/cli.go

```diff
@@ -43,6 +43,10 @@ func LoadMasterStateTryAll(masters []string, stateLoader func(ip, port string) (
 
 	if len(masters) > 0 {
 		leader, masters = masters[0], masters[1:]
+	} else {
+		err = errors.New("No masters given to fetch state.json")
+		logging.Error.Println(err)
```

Comment:
I don't think you should log here, it should be done by the caller based on the error you have created.